packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -135,34 +135,74 @@ function buildAndRun(args: Flags) {
     packageName,
   );
   const adbPath = getAdbPath();
-  if (args.deviceId) {
+  let devices;
```

Comment:
How about we ALWAYS get `const devices = adb.getDevices(adbPath)` and... `const device = args.deviceId ? selectSimulator(devices, args.deviceId) : undefined`

If `device` is undefined and `args.interactive` is `true`, we would proceed with the interactive logic.